kedro/extras/datasets/api/api_dataset.py

```diff
@@ -123,7 +124,15 @@ def _execute_request(self) -> requests.Response:
 
     def _load(self) -> Any:
         response = self._execute_request()
-        return response.json() if self._json else response.text
+        if not self._attribute:
+            output = response
+        elif self._attribute == "json":
+            output = response.json()
+        elif hasattr(response, self._attribute):
+            output = getattr(response, self._attribute)
+        else:
+            raise DataSetError("Response has no attribute: {}".format(self._attribute))
```

Comment:
Could just do a `try/except` on `getattr` directly here to be more pythonic, wdyt?